app/controllers/actions/CheckEmailVerificationFilter.scala

```diff
@@ -28,56 +29,89 @@ import services.{EmailVerificationService, SaveForLaterService}
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.http.HttpVerbs.GET
 import uk.gov.hmrc.play.http.HeaderCarrierConverter
+import utils.FutureSyntax.FutureOps
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class CheckEmailVerificationFilterImpl(mode: Option[Mode],
                                        frontendAppConfig: FrontendAppConfig,
                                        emailVerificationService: EmailVerificationService,
-                                       saveForLaterService: SaveForLaterService
+                                       saveForLaterService: SaveForLaterService,
+                                       registrationConnector: RegistrationConnector
                                       )(implicit val executionContext: ExecutionContext)
   extends ActionFilter[AuthenticatedDataRequest] with Logging {
 
   override protected def filter[A](request: AuthenticatedDataRequest[A]): Future[Option[Result]] = {
 
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
 
-    if (frontendAppConfig.emailVerificationEnabled && !mode.contains(AmendMode) && !mode.contains(RejoinMode)) {
+    if (frontendAppConfig.emailVerificationEnabled) {
       request.userAnswers.get(BusinessContactDetailsPage) match {
         case Some(contactDetails) =>
-          emailVerificationService.isEmailVerified(contactDetails.emailAddress, request.userId).flatMap {
-            case Verified =>
-              logger.info("CheckEmailVerificationFilter - Verified")
-              Future(Option.empty[Result])
-            case LockedTooManyLockedEmails =>
-              logger.info("CheckEmailVerificationFilter - LockedTooManyLockedEmails")
-              Future.successful(Option(Redirect(controllers.routes.EmailVerificationCodesAndEmailsExceededController.onPageLoad().url)))
-            case LockedPasscodeForSingleEmail =>
-              logger.info("CheckEmailVerificationFilter - LockedPasscodeForSingleEmail")
-              saveForLaterService.saveAnswers(
-                controllers.routes.EmailVerificationCodesExceededController.onPageLoad(),
-                Call(GET, request.uri)
-              )(request, executionContext, hc).map(Option(_))
-            case _ =>
-              logger.info("CheckEmailVerificationFilter - Not Verified")
-              Future.successful(Option(Redirect(controllers.routes.BusinessContactDetailsController.onPageLoad(NormalMode).url)))
+          if (mode.contains(AmendMode) || mode.contains(RejoinMode)) {
+            registrationConnector.getRegistration()(hc).flatMap {
```

Comment:
Don't need to get registration, it should already be in context of the request from 

https://github.com/hmrc/one-stop-shop-registration-frontend/pull/400/files#diff-aceaf3b69068fa36d5639f7bf613be54ca6038757c7e01b176c86037aa9996faR45

which is:

https://github.com/hmrc/one-stop-shop-registration-frontend/blob/main/app/models/requests/DataRequest.scala#L45

Reply:
Changes made